docker/daemon.go

```diff
@@ -81,7 +84,20 @@ func mainDaemon() {
 		return
 	}
 	eng := engine.New()
-	signal.Trap(eng.Shutdown)
+	sig.Trap(eng.Shutdown)
+	if *flDebug {
+		go func() {
+			sigs := make(chan os.Signal, 1)
+			signal.Notify(sigs, syscall.SIGQUIT)
+			buf := make([]byte, 16384)
+			for {
+				<-sigs
+				buf = buf[:runtime.Stack(buf, true)]
+				log.Debugf("=== BEGIN Docker daemon goroutine stacktrace ===\n\n")
```

Comment:
I'm not sure if `\n\n` is necessary here, looks like `log.Debugf` should add EOL itself.